adagrams/game.py

```diff
@@ -1,11 +1,159 @@
+import random
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+
+
 def draw_letters():
-    pass
+
+    my_list = [] #creating an empty list to collect randomly picked letter
+
+    #converting dictionary keys into list representing the 'count' times of letter and storing in pool_list
+    pool_list = []
+    for letter, count in LETTER_POOL.items():
+        pool_list += [letter] * count # add letter 'count' times to pool_list
+        
+    while len(my_list) < 10:
+        random_key = random.randint(0, len(pool_list) - 1) #randomly select index from pool_list
+        
+        random_letter = pool_list[random_key] #getting letter associated with index
+        pool_list.remove(random_letter) #removing letter from pool_list
+        my_list.append(random_letter) #adding this letter to my_list
+
+    return my_list
+
+
+letter_bank = draw_letters() # retreiving my_list of 10 listter from draw_letter function
+
+word = "SANTAAA"
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+
+    # getting user input as string 
+    upper_word = word.upper()
+
+    letter_bank_dict = {} #empty dictionary to store 
+    inputed_word_dict = {} #empty dictionary to store letter and frequency from user
+
+
+    # adding upper_word into inputed_word_dict with letter: count format
+    for letter in upper_word:
```

Comment:
Prefer to call `upper` directly in the for loop so you can remove line 59 completely. 

```python
for letter in word.upper():
    # rest of your code
```